text/entities/0256-entities-data-model.md

```diff
@@ -0,0 +1,790 @@
+
+When the entity's state is changed it is expected that the source will
+emit a new EntityState event with a fresh timestamp and full list of
+values of all other fields.
+
+Entity event producers are recommended to periodically emit events even
+if the entity does not change. In this case the Type, Id and Attribute
+fields will remain the same, but a fresh Timestamp will be recorded in
+the event. Producing such events allows the system to be resilient to
+event losses. Even if some events are lost eventually the correct state
+of the entity is more likely to be delivered to the final destination.
+Periodic sending of EntityState events also serves as a liveliness
+indicator (see below how it can be used in lieu of EntityDelete event).
+
+### EntityDelete Event
+
+EntityDelete event indicates that a particular entity is gone:
+
+<table>
+   <tr>
+    <td><strong>Field</strong>
+    </td>
+    <td><strong>Type</strong>
+    </td>
+    <td><strong>Description</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Timestamp
+    </td>
+    <td>nanoseconds
+    </td>
+    <td>The time when the entity was deleted. The time is measured by
+the origin clock. The field is required.
+    </td>
+   </tr>
+   <tr>
+    <td>Type
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+   <tr>
+    <td>Id
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+</table>
+
+Note that EntityDelete is optional and is not guaranteed to be sent when
+the entity is gone. Recipients of entity signals should be prepared to
+handle this situation by expiring entities that are no longer seeing
+EntityState events reported (i.e. treat the presence of EntityState
+events as a liveliness indicator).
+
+The expiration mechanism is based on the previously reported `Interval` field of
+EntityState event. The recipient can use this value to compute when to expect the next
+EntityState event and if the event does not arrive in a timely manner (plus some slack)
+it can consider the entity to be gone even if the EntityDelete event was not observed.
+
+## Entity Identification
+
+_This section is a supplementary guideline and is not part of logical data model._
+
+The data model defines the structure of the entity Id field. This section explains
+how the Id field is computed.
+
+### LID, GID and IDCONTEXT
+
+All entities have a local ID (LID) and a global ID (GID).
+
+The LID is unique in a particular identification context, but is not necessarily globally
+unique. For example a process entity's LID is its PID number and process start time.
+The (PID,StartTime) pair is unique only in the context of a host where the process runs
+(and the host in this case is the identification context).
+
+The GID of an entity is globally unique, in the sense that for the entire set of entities
+in a particular telemetry store no 2 entities exist that have the same GID value.
+
+The GID of an entity E is defined as:
+
+`GID(E) = { LID(E), GID(IDCONTEXT(E)) }`
```

Comment:
Can we use the word union instead of `{}`. I'm not aware of that being a standard syntax.

```suggestion
`GID(E) = UNION( LID(E), GID(IDCONTEXT(E)) )`
```

Reply:
Fixed.